### PR TITLE
SEN-2590 Add recipes for migrating Hamcrest size assertions to AssertJ

### PR DESCRIPTION
**Pull request by bbogason**

### CODE REVIEW

File: recipes/Java/AssertJ/From Hamcrest/Hamcrest_hasSize.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+                  type: org.hamcrest.collection.IsArrayWithSize
+                - name: aMapWithSize
+                  type: org.hamcrest.Matchers
+                - name: aMapWithSize
+                  type: org.hamcrest.collection.IsMapWithSize
+      argCount: 3
+    type: org.hamcrest.MatcherAssert
+availableFixes:
+- doStaticImports: true
+  name: Change to assertThat(actual).isEqualTo(expected)
```
Comment (bbogason) [inline recipes/Java/AssertJ/From Hamcrest/Hamcrest_hasSize.yaml:65]:
```suggestion
  name: Change to assertThat(actual).hasSize(expectedSize)
```

Similar elsewhere

File: recipes/Java/AssertJ/From Hamcrest/Hamcrest_hasSize.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  availableIf:
+    markedElement:
+      is:
+        methodcall:
+          argCount: 2
+  actions:
+  - rewrite:
+      to: org.assertj.core.api.Assertions.assertThat({{{ arguments.0 }}}).hasSize({{{ arguments.1.arguments.0 }}})
+- doStaticImports: true
+  name: Change to assertThat(actual).as(reason).isEqualTo(expected)
```
Comment (camac) [inline recipes/Java/AssertJ/From Hamcrest/Hamcrest_hasSize.yaml:75]:
```suggestion
  name: Change to assertThat(actual).as(reason).hasSize(expectedSize)
```

File: recipes/Java/AssertJ/From Hamcrest/Hamcrest_isHasSize.yaml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                name: is
+                type:
+                  reference:
+                    matches: org\.hamcrest\.(CoreMatchers|Matchers|core\.Is)
+                  checkInheritance: true
+      argCount: 3
+    type: org.hamcrest.MatcherAssert
+availableFixes:
+- doStaticImports: true
+  name: Change to assertThat(actual).isEqualTo(expected)
```
Comment (camac) [inline recipes/Java/AssertJ/From Hamcrest/Hamcrest_isHasSize.yaml:85]:
```suggestion
  name: Change to assertThat(actual).hasSize(expectedSize)
```

File: recipes/Java/AssertJ/From Hamcrest/Hamcrest_isHasSize.yaml [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  availableIf:
+    markedElement:
+      is:
+        methodcall:
+          argCount: 2
+  actions:
+  - rewrite:
+      to: org.assertj.core.api.Assertions.assertThat({{{ arguments.0 }}}).hasSize({{{ arguments.1.arguments.0.arguments.0 }}})
+- doStaticImports: true
+  name: Change to assertThat(actual).as(reason).isEqualTo(expected)
```
Comment (camac) [inline recipes/Java/AssertJ/From Hamcrest/Hamcrest_isHasSize.yaml:95]:
```suggestion
  name: Change to assertThat(actual).as(reason).hasSize(expectedSize)
```

**Review by camac — CHANGES_REQUESTED** [review]
I like how you have covered so many situations with just 2 recipes!

The only changes required are the updates to the fix names, I have added the other 3 suggested changes to the fix names (following on from your first suggested change).

**Review by wanderneto — APPROVED** [review]
They looked fine. Maybe just change the fixes names as @camac suggested, then the name will reflect exactly which method will be used.

**Comment by bbogason** [comment]
> I like how you have covered so many situations with just 2 recipes!

Yes...
- I find Hamcrest a bit infuriating because there are so many ways of achieving the same thing.
- Ideally I would have only written a single recipe, but that lead to a recipe that even I could no longer follow or possibly maintain.

**Pull request merged**